src/Session.hs

```diff
@@ -125,6 +134,40 @@ sessionRestart session@Session{..} = do
     sessionStart session cmd setup
 
 
+performEvals :: Ghci -> Bool -> [FilePath] -> IO [Load]
+performEvals _ False _ = pure []
+performEvals ghci True reloaded = do
+  cmds <- traverse getCommands reloaded
+  fmap join $ flip traverse cmds $ \(file, cmds') ->
+    flip traverse cmds' $ \(num, cmd) -> do
+      ref <- newIORef []
+      execStream ghci (intercalate " " $ lines cmd) $ \_ resp -> modifyIORef ref (++ resp)
```

Comment:
This is probably an _O(n^2)_ algorithm. I'd suggest `modifyIORef ref (resp:)` and then after the `readIORef` doing a `concat . reverse`